### PR TITLE
NAS-120252 / 23.10 / Allow specifying acls for apps

### PR DESCRIPTION
**Pull request by Qubad786**
## Context

`filesystem.add_to_acl` has been added to allow app devs to correctly specify ACLs for host paths. A feature `normalize/acl` has been added which app devs can specify to consume the new feature.

### CODE REVIEW

**Review by yocalebo — CHANGES_REQUESTED** [review]

File: src/middlewared/middlewared/plugins/chart_releases_linux/normalize_schema.py
Comment (yocalebo) [reply src/middlewared/middlewared/plugins/chart_releases_linux/normalize_schema.py:18]: We should definitely fix the `REF_MAPPING` dict. It's confusing to have 2 different versions of spellings.

Comment (Qubad786) [reply src/middlewared/middlewared/plugins/chart_releases_linux/normalize_schema.py:18]: @yocalebo we cannot change the refs as they are already in production and being used by app devs..
I'll make the changes to function names to consume the US version

Comment (yocalebo) [reply src/middlewared/middlewared/plugins/chart_releases_linux/normalize_schema.py:18]: That's painful...but I guess something we gotta live with now.

File: src/middlewared/middlewared/plugins/chart_releases_linux/normalize_schema.py
Comment (yocalebo) [inline src/middlewared/middlewared/plugins/chart_releases_linux/normalize_schema.py:171]: Just a preference but I like to be explicit with checks. i.e.

```suggestion
        if (action_dict := next((d for d in context['actions'] if d['method'] == 'apply_acls'), None)) is not None:
```

File: src/middlewared/middlewared/plugins/chart_releases_linux/acl.py
Comment (anodos325) [inline src/middlewared/middlewared/plugins/chart_releases_linux/acl.py:15]: There are a few cases where this may happen:
1) Invalid payload
2) Permissions on the parent path would prevent the ACL from granting intended permissions (e.g. parent path has mode of 0700)
3) Dataset has ACLs disabled
4) Dataset has POSIX1E ACL type and payload specifies FULL_CONTROL

It would be helpful to give administrator some feedback here.

**Comment by anodos325** [comment]
There's CI coverage of the API endpoint, but it would be good to have some tests for this area as well.

**Review by yocalebo — APPROVED** [review]
I see no apparent issues but wait for @anodos325 final review before merge, please.

**Review by anodos325 — APPROVED** [review]

**Review by sonicaj — APPROVED** [review]
Good work!

**Pull request merged**
